tests/Backend.Tests/Auth/BaseOAuthAuthenticationTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Jpp.Common.Backend.Auth;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+
+namespace Jpp.Common.Backend.UnitTests.Auth
+{
+    [TestFixture]
+    class BaseOAuthAuthenticationTests
+    {
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            Backend.BASE_URL = "jppuk.net";
+        }
+
+        [Test]
+        public async Task OnAuthentication_Called()
+        {
+            var handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            handlerMock
+                .Protected()
+                .Setup<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.IsAny<HttpRequestMessage>(),
+                    ItExpr.IsAny<CancellationToken>()
+                )
+                .ReturnsAsync(new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent("[{'id':1,'value':'1'}]"),
+                })
+                .Verifiable();
+
+            bool eventFired = false;
+
+            IAuthentication auth = new BaseOAuthAuthenticationStub(handlerMock.Object);
+            auth.AuthenticationChanged += new EventHandler(delegate (Object o, EventArgs a) { eventFired = true; });
+            await auth.Authenticate();
+
+            Assert.IsTrue(eventFired);
+        }
+
+        [Test]
+        public async Task GetUserProfile_Valid()
```

Comment:
I'm assuming you'll add later tests for !Authenticated throwing NotAuthenticatedException and also invalid user claims?

Reply:
Good point, no. Ill add these now